src/builder/builder.js

```diff
@@ -62,10 +62,7 @@ module.exports = function (createChunker, ipldResolver, createReducer, _options)
     waterfall([
       (cb) => DAGNode.create(d.marshal(), cb),
       (node, cb) => {
-        ipldResolver.put({
-          node: node,
-          cid: new CID(node.multihash)
-        }, (err) => cb(err, node))
+        ipldResolver.put(node, new CID(node.multihash), (err) => cb(err, node))
```

Comment:
It has to support two variants -- https://github.com/ipld/js-ipld-resolver#putnode-cid--format-hashalg-callback --, how would you like for this to be?